src/main/java/igrad/logic/commands/requirement/RequirementAssignCommand.java

```diff
@@ -0,0 +1,90 @@
+import igrad.logic.commands.exceptions.CommandException;
+import igrad.model.Model;
+import igrad.model.module.Module;
+import igrad.model.module.ModuleCode;
+import igrad.model.requirement.Credits;
+import igrad.model.requirement.Name;
+import igrad.model.requirement.Requirement;
+
+/**
+ * Assigns modules under a particular requirement.
+ */
+public class RequirementAssignCommand extends RequirementCommand {
+    public static final String COMMAND_WORD = REQUIREMENT_COMMAND_WORD + "assign";
+
+    public static final String MESSAGE_DETAILS = COMMAND_WORD + ": Edits the requirement. "
+        + "Existing requirement will be overwritten by the new name and/or credits.\n";
+
+    public static final String MESSAGE_USAGE = "Parameter: "
+        + "[" + PREFIX_NAME + "NEW_NAME] "
+        + "[" + PREFIX_CREDITS + "NEW_CREDITS]\n"
+        + "Example: " + COMMAND_WORD + " Unrestrained Elves "
+        + PREFIX_NAME + "Unrestricted Electives";
+
+    public static final String MESSAGE_REQUIREMENT_NO_MODULES = "There must be at least one modules assigned.";
+
+    public static final String MESSAGE_MODULES_NON_EXISTENT =
+        "Not all Modules exist in the system. Please try other modules.";
+
+    public static final String MESSAGE_MODULES_ALREADY_EXIST_IN_REQUIREMENT =
+        "Some Modules already exists in this requirement. Please try other modules.";
+    public static final String MESSAGE_REQUIREMENT_ASSIGN_MODULE_SUCCESS = "Modules assigned under Requirement: %1$s";
+
+    private Name requirementName;
+    private List<ModuleCode> moduleCodes;
+
+    public RequirementAssignCommand(Name requirementName, List<ModuleCode> moduleCodes) {
+        requireAllNonNull(requirementName, moduleCodes);
+
+        this.requirementName = requirementName;
+        this.moduleCodes = moduleCodes;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // Retrieve the requirement in question that we want to assign modules under..
+
+        // First check if the requirement exists in the course book
+        Requirement requirementToAssign = model.getRequirementByName(requirementName)
+            .orElseThrow(() -> new CommandException(MESSAGE_REQUIREMENT_NON_EXISTENT));
+
+        Requirement editedRequirement = new Requirement(requirementToAssign);
+
+        final List<Module> modulesToAssign = model.getModulesByModuleCode(moduleCodes);
+
+        // First check, if all modules (codes) are existent modules in the course book (they should all be)
+        if (modulesToAssign.size() < moduleCodes.size()) {
+            throw new CommandException(MESSAGE_MODULES_NON_EXISTENT);
+        }
+
+        // Now check, if any modules specified are existent in the requirement (they should not)
+        if (editedRequirement.hasModule(modulesToAssign)) {
+            throw new CommandException(MESSAGE_MODULES_ALREADY_EXIST_IN_REQUIREMENT);
```

Comment:
actually right this part no need to throw exception, i thinj can just add the modules that are not yet added but should probably have a user feedback to tell them the list of modules added (though idk if its necessary)

Reply:
ok this part I don't quite understand

Reply:
for the 'adding of modules which are not yet added', is it when  i try to assign CS1101S, CS2030 to a requirement; say UE, and suppose that CS1101S is already under this requirement, then the system only adds CS2030 (instead of flagging error)?

Reply:
then as for the feedback part, hmm maybe we could have that (currently it just prints out how many credits fulfilled and required, etc).

Reply:
yes instead of throwing an error, just add the ones that are not already added!

Reply:
hmm ok i'll try to implement that!